src/index.ts

```diff
@@ -38,7 +38,23 @@ const prepVercel = async () => {
   console.log("⚡️ Installing 'vercel' CLI...");
   console.log("⚡️");
 
-  const vercelBuild = spawn("npm", ["install", "-D", "vercel"]);
+  const pkgManager = await getPkgManager();
+  let vercelBuild;
+
+  switch (pkgManager) {
+    case "pnpm":
+      console.log('⚡️ Detected "pnpm", running pnpm add -D vercel');
+      console.log("⚡️");
+      vercelBuild = spawn("pnpm", ["add", "-D", "vercel"]);
+      break;
+    case "yarn":
+      console.log('⚡️ Detected "yarn", running yarn add -D vercel');
+      console.log("⚡️");
+      vercelBuild = spawn("yarn", ["add", "-D", "vercel"]);
+      break;
+    default:
+      vercelBuild = spawn("pnpm", ["add", "-D", "vercel"]);
```

Comment:
Default case would be unaltered as I understand.
```suggestion
      vercelBuild = spawn("npm", ["install", "-D", "vercel"]);
```